lib/truffle/rubygems/installing_extra_executables.rb

```diff
@@ -0,0 +1,38 @@
+module Gem
+  module InstallingExtraExecutables
+    extend self
+
+    def install_hooks_for(*extra_bin_dirs)
+      install_hook   = -> tool { hook(extra_bin_dirs, tool, :generate_bin) }
+      uninstall_hook = -> tool { hook(extra_bin_dirs, tool, :remove_executables, tool.spec) }
+      Gem.post_install_hooks.push(install_hook)
+      Gem.pre_uninstall_hooks.push(uninstall_hook)
+      [install_hook, uninstall_hook]
```

Comment:
Since the return value is not used, should this be removed?

Reply:
It's required to be able to remove the hooks with remove_hooks method.